src/pages/tokenDrop/Footer.tsx

```diff
@@ -1,41 +1,70 @@
 import { Box, useTheme } from "@mui/material";
 import { Link } from "elements";
-import { FunctionComponent } from "react";
+import { Modal } from "components";
+import { FunctionComponent, useState } from "react";
+import SocialsModal from "./SocialsModal";
+import FAQModal from "./FAQModal";
 
 interface Links {
-  readonly to: string;
+  readonly key: string;
   readonly label: string;
+  readonly to: string;
   readonly type: "router" | "web";
 }
 
 const Footer: FunctionComponent = () => {
   const theme = useTheme();
+  const [isOpen, setIsOpen] = useState(false);
+  const [modalId, setModalId] = useState("");
+
+  const handleClose = () => setIsOpen(false);
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  const handleOpen = (event: any) => {
+    setModalId(event?.target?.getAttribute("id"));
```

Comment:
So the reason why i am doing this is to get the id attribute of the link that was clicked so that i can set the modal content depending on that. I feel like there's a better way to do it by using `useRef` but it was giving me an error about forwarding ref, this is something we can look at in the future